File: hmh-frontend/src/api/summary.ts
import client from "./client";

export interface DailySummary {
  date_today: string;
  spend_today: number;
  spend_week: number;
  spend_month: number;
  deliveries_today: number;
  active_lots: number;
  delayed_lots: number;
  completed_lots: number;
  open_alerts: number;
  material_overrun_alerts: number;
  unmatched_invoices: number;
  vehicle_spend_today: number;
  vehicle_spend_month: number;
  burn_rate_daily_avg: number;
}

export const summaryApi = {
  daily: async (projectId?: string): Promise<DailySummary> => {
    const params = projectId ? { project_id: projectId } : {};
    const res = await client.get<{ data: DailySummary }>("/summary/daily", { params });
    return res.data.data;
  },
};

File: hmh-backend/app/api/v1/summary.py
"""Owner daily summary route."""

import uuid
from typing import Optional

from fastapi import APIRouter, Query

from app.dependencies import ALL_ROLES, DbSession
from app.schemas.common import ApiSuccess
from app.services import summary_service
from pydantic import BaseModel
from datetime import date


router = APIRouter(prefix="/summary", tags=["summary"])


class DailySummaryRead(BaseModel):
    date_today: date
    spend_today: float
    spend_week: float
    spend_month: float
    deliveries_today: int
    active_lots: int
    delayed_lots: int
    completed_lots: int
    open_alerts: int
    material_overrun_alerts: int
    unmatched_invoices: int
    vehicle_spend_today: float
    vehicle_spend_month: float
    burn_rate_daily_avg: float


@router.get("/daily", response_model=ApiSuccess[DailySummaryRead], dependencies=[ALL_ROLES])
def get_daily_summary(
    db: DbSession,
    project_id: Optional[uuid.UUID] = Query(None),
):
    summary = summary_service.get_daily_summary(db, project_id)
    return ApiSuccess(data=DailySummaryRead(**vars(summary)))
